IdokladSdk/Validation/Attributes/IdentificationNumberAttribute.cs

```diff
@@ -10,15 +9,62 @@ namespace IdokladSdk.Validation.Attributes
     /// </summary>
     public class IdentificationNumberAttribute : ValidationAttribute
     {
+        private readonly string _hasNoIdentificationNumberProperty;
+
+        public IdentificationNumberAttribute(string hasNoIdentificationNumber = null)
```

Comment:
Asi úplně nerozumím, proč tady tomu konstruktoru předáváš jméno té property a ne rovnou hodnotu té property HasNoIdentificationNumber. Kdyby to bralo rovnou to bool hodnotu, nemusela by existovat metoda GetHasNoIdentificationNumber (33) a i If statement by byl jednodušší (24). Navíc teď když je ta stringová hodnota název té property, neodpovídá tomu název toho argumentu. Od hasNoIdentificationNumber bych čekal typ bool ne string.

Reply:
Protože v konstruktoru je název property, jejíž hodnotu máš až v runtime v metodě IsValid. Do parametrů attributu lze vložit jen konstanty. Vložit přímo bool hodnotu by bylo lepší, ale nejde to.